### PR TITLE
Educational part

### PR DESCRIPTION
**Pull request by Mary0110**
Added elastic search with custom analyser settings and docker support for all microservices

### CODE REVIEW

File: Services/Edu/PsychologicalSupportPlatform.Edu.Application/Interfaces/IEduMaterialService.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace PsychologicalSupportPlatform.Edu.Application.Interfaces;
+
+public interface IEduMaterialService
+{
+    Task<int> UploadEduMaterialAsync(AddEduMaterialDTO dto, CancellationToken token);
+    
+    Task<MemoryStream> DownloadEduMaterialAsync(int id, CancellationToken token);
+    
+    Task<List<EduMaterialDTO>> GetEduMaterialsByStudentAsync(int studentId, int pageNumber, int pageSize, CancellationToken token);
```
Comment (katyalys) [inline Services/Edu/PsychologicalSupportPlatform.Edu.Application/Interfaces/IEduMaterialService.cs:12]: create model for pageSize, pageNumber, studentId

File: Services/Edu/PsychologicalSupportPlatform.Edu.Application/Interfaces/IEduMaterialService.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    Task<List<EduMaterialDTO>> GetAllEduMaterialsAsync(int pageNumber, int pageSize);
+
+    Task AddEduMaterialToStudentAsync(AddEduMaterialToStudentDTO dto, CancellationToken token);
+    
+    Task<IEnumerable<EduMaterialDTO>> SearchAsync(string text, int pageNumber, int pageSize, CancellationToken cancellationToken);
```
Comment (katyalys) [inline Services/Edu/PsychologicalSupportPlatform.Edu.Application/Interfaces/IEduMaterialService.cs:18]: same

File: Services/Edu/PsychologicalSupportPlatform.Edu.Application/Interfaces/IStudentHasEduMaterialRepository.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+using PsychologicalSupportPlatform.Common.Repository;
+using PsychologicalSupportPlatform.Edu.Domain.Entities;
+
+namespace PsychologicalSupportPlatform.Edu.Application.Interfaces;
+
+public interface IStudentHasEduMaterialRepository : ISQLRepository<StudentHasEduMaterial>
+{
+    Task<List<EduMaterial>> GetEduMaterialsByStudentAsync(int studentId, int pageNumber, int pageSize);
```
Comment (katyalys) [inline Services/Edu/PsychologicalSupportPlatform.Edu.Application/Interfaces/IStudentHasEduMaterialRepository.cs:8]: same

File: Services/Edu/PsychologicalSupportPlatform.Edu.Application/Services/TestService.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        }
+        
+        var testHasStudent = _mapper.Map<UserTestResult>(dto);
+        var added = await _userTestResultRepository.AddAsync(testHasStudent);
+        await _userTestResultRepository.SaveAsync();
+
+        var answerRequests = _mapper.Map<List<QuestionResult>>(dto.AnswerRequestDTO.QuestionResultDTOs);
+        
+        foreach (var entity in answerRequests)
+        {
```
Comment (katyalys) [inline Services/Edu/PsychologicalSupportPlatform.Edu.Application/Services/TestService.cs:50]: use linq foreach method

File: Services/Edu/PsychologicalSupportPlatform.Edu.Application/Services/TestService.cs [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        for(var i = 0; i < questionList.Count; i++)
+        {
+            var el = questionList[i];
+            el.TestId = added.Id;
+            var addedQuestion = await _questionRepository.AddAsync(el);
+            await _questionRepository.SaveAsync();
+            var answerList = _mapper.Map<List<Answer>>(addProductDto.Questions[i].Answers);
+            
+            foreach (var answer in answerList)
+            {
```
Comment (katyalys) [inline Services/Edu/PsychologicalSupportPlatform.Edu.Application/Services/TestService.cs:95]: same

**Review by katyalys — CHANGES_REQUESTED** [review]

File: Services/Edu/PsychologicalSupportPlatform.Edu.Application/Interfaces/IEduMaterialService.cs (hunk 2 above)
Comment (Mary0110) [reply Services/Edu/PsychologicalSupportPlatform.Edu.Application/Interfaces/IEduMaterialService.cs:18]: fixed

File: Services/Edu/PsychologicalSupportPlatform.Edu.Application/Interfaces/IStudentHasEduMaterialRepository.cs (hunk 3 above)
Comment (Mary0110) [reply Services/Edu/PsychologicalSupportPlatform.Edu.Application/Interfaces/IStudentHasEduMaterialRepository.cs:8]: fixed

File: Services/Edu/PsychologicalSupportPlatform.Edu.Application/Services/TestService.cs (hunk 4 above)
Comment (Mary0110) [reply Services/Edu/PsychologicalSupportPlatform.Edu.Application/Services/TestService.cs:50]: fixed

File: Services/Edu/PsychologicalSupportPlatform.Edu.Application/Services/TestService.cs (hunk 5 above)
Comment (Mary0110) [reply Services/Edu/PsychologicalSupportPlatform.Edu.Application/Services/TestService.cs:95]: fixed

**Review by katyalys — APPROVED** [review]

**Pull request merged**